src/awx_plugins/credentials/azure_kv.py

```diff
@@ -71,24 +88,52 @@
     ],
     'required': [
         'url',
-        'client',
-        'secret',
-        'tenant',
         'secret_field',
     ],
 }
 
 
-def azure_keyvault_backend(**kwargs):
-    csc = ClientSecretCredential(
-        tenant_id=kwargs['tenant'],
-        client_id=kwargs['client'],
-        client_secret=kwargs['secret'],
-    )
-    kv = SecretClient(credential=csc, vault_url=kwargs['url'])
-    return kv.get_secret(
-        name=kwargs['secret_field'],
-        version=kwargs.get('secret_version', ''),
+def azure_keyvault_backend(
+        url: str,
+        client: str,
+        secret: str,
+        tenant: str,
+        secret_field: str,
+        secret_version: str,
+) -> str:
+    """Get a credential and retrieve a secret from an Azure Key Vault.
+
+    :param url: An Azure Key Vault URI.
+    :param client: The Client ID  (optional).
+    :param secret: The Client Secret  (optional).
+    :param tenant: The Tenant ID  (optional).
+    :param secret_field: The name of the secret to retrieve from the
+        vault.
+    :param secret_version: The version of the secret to retrieve
+        (optional).
+    :returns: The secret from the Key Vault.
+    :raises RuntimeError: If the software is not being run on an Azure
+        VM.
+    """
+    try:
+        if '' in [tenant, client, secret]:
+            cred = ManagedIdentityCredential()
```

Comment:
Since only this line is expected to raise that exception, move the try-block to only wrap this line but not others.